pkg/crds/controllers/batch/batchjob_controller_helpers.go

```diff
@@ -185,6 +186,34 @@ func (r *BatchJobReconciler) enqueuePayload(ctx context.Context, batchJob batch.
 	return nil
 }
 
+func (r *BatchJobReconciler) createWorkerConfigMap(ctx context.Context, batchJob batch.BatchJob) error {
+	probesSpecData, err := workloads.GenerateProbesConfigMapData(batchJob.Spec.Probes)
```

Comment:
Consider this design:

```go
jobSpec, err := ConvertControllerBatchToJobSpec(batchJob)
jobSpecEncoded, err := json.MarshalIndent(taskJobSpec, "", "\t")
probesEncoded, err := json.MarshalIndent(probes, "", "\t")

configMapData := map[string]string{
  "job_spec.json": string(jobSpecEncoded),
  "probes.json": string(probesEncoded),
}
```

Another option (pseudocode):

```go
type configMapFields struct {
	BatchJobSpec *BatchJobSpec
	TaskJobSpec *TaskJobSpec
	Probes map[string]Probe
}

func GenerateConfigMap(configMapFields) {
  // check if not nil on each one
}
```

Reply:
I think the 2nd option is the better alternative. Let's discuss this.